src/helpers/compose.ts

```diff
@@ -1,7 +1,19 @@
-export const compose = <T>(...fns: Function[]) => (x: T): T =>
-  fns.reduceRight((v, f) => f(v), x);
+type Func<T, R> = (a: T) => R;
 
-export const pipe = <T>(...fns: Function[]) => (x: T): T =>
-  fns.reduce((v, f) => f(v), x);
+export function compose<F extends Function>(f: F): F;
```

Comment:
4 exports of compose function? how does it work?

Reply:
It'a function overloading, you can look this up here: https://www.typescriptlang.org/docs/handbook/functions.html#overloads
You can expose different typings depending on the arguments or it's number. Here I'm using it to have a better return type inference.

Reply:
For eg here: `src/services/updatePackageJson.ts` I don't have to explicitly pass input to the generic function as it could be easily inference from existing arguments

Reply:
nice thing! :)